pickhardtpayments/Attempt.py

```diff
@@ -0,0 +1,120 @@
+    PLANNED = 1
+    INFLIGHT = 2
+    ARRIVED = 3
+    FAILED = 4
+    SETTLED = 5
+
+
+class Attempt:
+    """
+    An Attempt describes a path (a set of channels) of an amount from sender to receiver.
+
+    When sending an amount of sats from sender to receiver, a payment is usually split up and sent across
+    several paths, to increase the probability of being successfully delivered. Each of this path is referred to as an
+    Attempt.
+    An Attempt consists of a list of Channels (class:Channel) and the amount in sats to be sent through this path.
+
+    :param path: a list of Channel objects from sender to receiver
+    :type path: list[Channel]
+    :param amount: the amount to be transferred from source to destination
+    :type amount: int
+    """
+
+    def __init__(self, path: list[Channel], amount: int = 0):
+        """Constructor method
+        """
+        self._routing_fee = None
+        self._probability = None
+        self._path = path
+        self._status = AttemptStatus.PLANNED
+        self._amount = amount
+
+    def __str__(self):
+        description = "Path with {} channels to deliver {} sats and status {}.".format(len(self._path),
+                                                                                       self._amount, self._status.name)
+        if self._routing_fee > 0:
+            description += "\nsuccess probability of {:6.2f}% , fee of {:8.3f} sat and a ppm of {:5} ".format(
+                self._probability * 100, self._routing_fee/1000, int(self._routing_fee * 1000 / self._amount))
+        return description
+
+    @property
+    def path(self) -> list[Channel]:
+        """Returns the path of the attempt.
+
+        :return: the list of Channels that the path consists of
+        :rtype: list[Channel]
+        """
+        return self._path
+
+    @property
+    def amount(self) -> int:
+        """Returns the amount of the attempt.
+
+        :return: the amount that was tried to send in this Attempt
+        :rtype: int
+        """
+        return self._amount
+
+    @property
+    def status(self) -> AttemptStatus:
+        """Returns the status of the attempt.
+
+        :return: returns the state of the attempt
+        :rtype: AttemptStatus
+        """
+        return self._status
+
+    @status.setter
+    def status(self, value: AttemptStatus):
+        """Sets the status of the attempt.
+
+        A flag to describe if the path failed, succeeded or was used for settlement (see enum SettlementStatus)
+
+        :param value: Current state of the Attempt
+        :type value: AttemptStatus
+        """
+        self._status = value
+
+    @property
+    def routing_fee(self) -> int:
+        """Returns the accrued routing fee in msat requested for this path.
+
+        :return: accrued routing fees for this attempt in msat
+        :rtype: int
+        """
+        return self._routing_fee
+
+    @routing_fee.setter
+    def routing_fee(self, value: int):
```

Comment:
an Attempt changes probabilities in the Uncertainty Network, thus consecutive attempts lead to new uncertainties.
Calculation needs to be done in attempt class with API from Uncertainty Network (given a path, what are fees and uncertainties)

TODO: 
- pass UncertaintyNetwork to Attempt Class
- add amount to planned amount in uncertainty network so that probabilities are correct 
- But Uncertainty Network hasn't PLANNED status currently, only INFLIGHT. 
Two Options:
  1. planned moves to inflight, when onion arrives but is not yet settled.
  2. using inflight is ok, as it produces the accurate probabilities.

Next steps: go with option 2 and see later, if refactoring in useful